### PR TITLE
Adds script to publish site to GitHub

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Adds script to publish site to GitHub

### Checklist
- [ x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
Would this be meant for people who want to build the site from their own forks?

File: build_gh_pages.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+bundle exec jekyll build --config _config.yml,_config_temp.yml --destination ${build_dir} --incremental
+
+# Delete the temp config file
+rm _config_temp.yml
+
+cd ${build_dir}
+touch .nojekyll
+git init
+git remote add origin "git@github.com:${gh_username}/${gh_username}.github.io.git"
```
Comment (AMoo-Miki) [inline build_gh_pages.sh:39]: We should make the repo configurable as it is likely that the default repo is in use for their personal use.

File: build_gh_pages.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      ?) function_help;;
+   esac
+done
+
+if [ -z "${gh_username}" ]; then 
+    echo 'GitHub username is required.'
+    function_help
+fi
+
+build_dir=$(mktemp -d -t build-XXXX)
```
Comment (AMoo-Miki) [inline build_gh_pages.sh:22]: This needs to be cross-platform.

**Review by AMoo-Miki — COMMENTED** [review]
We should make this cross-platform.

**Comment by kolchfa-aws** [comment]
Closing because of the limited use case of this script.

**Pull request closed without merging**